transformations/negate_strengthen/transformation.py

```diff
@@ -0,0 +1,456 @@
+                # if method is nor format
+                if alt_text is not None:
+                    if method[-1]=='JJ_1_2':
+                        alt_text, alt_word_2 = self.improve_negation_flow(alt_text.copy(), edit_id+2, method)
+                        alt_word = [alt_word, alt_word_2]
+            else:
+                alt_word = None
+                alt_text = None
+
+            # keep only successful edits
+            if None not in method:
+                t_dict[jx] = {
+                    'edit_text': edit_text,
+                    'edit_id': edit_id,
+                    'edit_method': method,
+                    'alt_word': alt_word,
+                    'alt_text': alt_text
+                }
+
+        return t_dict
+
+
+    def text_to_stronger_edits(self, _text, sentid2tid):
+        # clean multiple whitespaces
+        _text = _RE_COMBINE_WHITESPACE.sub(" ", _text).strip()
+        text = _text.split(' ')
+
+        # get weaker words for converting
+        table = self.text_to_spacy_table_info(_text)
+        get_roots = table[table['lemma'].isin(MODAL_STRENGTHEN_DICT.keys())].index
+        itemdict = {'text': [(int(idx+1),0,int(idx),t) for idx, t in enumerate(text)]}
+
+        t_dict = {}
+        for jx, root in enumerate(get_roots):
+            
+            if table.loc[root+1,'lemma']=='be':
+                text[root] = "<edit>was</edit>"
+                text[root+1] = ""
+                method = ['MOD_2_1']
+            elif  table.loc[root+1,'lemma']=='have':
+                if table.loc[root+2,'lemma']=='be':
+                    text[root] = "<edit>was</edit>"
+                    text[root+1] = ""
+                    text[root+2] = ""
+                    method = ['MOD_3_2']
+                else:
+                    text[root] = "<edit>had</edit>"
+                    text[root+1] = ""
+                    method = ['MOD_3_1'] 
+            elif (table.loc[root,'pos_tag']=='MD') and (table.loc[root+1,'pos_tag']=='RB'):
+                text[root] = '<edit>'+ MODAL_STRENGTHEN_DICT[text[root]] + '</edit>'
+                text[root+1] = ""
+                # text[root+2] = conjugate_en(verb=text[root+2],tense='past',number='singular')
+                method = ['MOD_4_1']
+            else:
+                text[root] = '<edit>'+ MODAL_STRENGTHEN_DICT[text[root]] + '</edit>'
+                method = ['MOD_1_1']
+            
+            edit_text = text
+            edit_id = root
+
+            # keep only successful edits
+            if None not in method:
+                t_dict[jx] = {
+                    'edit_text': edit_text,
+                    'edit_id': edit_id,
+                    'edit_method': method,
+                    'alt_word': None,
+                    'alt_text': None
+                }
+
+        return t_dict
+
+
+    def generate(self, sentence: str, target: str):
```

Comment:
Oh, I understand why there's a signal_id in the test case. Maybe you can update the sentence type hint to `Union[dict, str]` so it would be clearer.

Reply:
Ok, done!